Fourth

solmoonkang opened the pull request:
userInfo API 구현
updatePassword API 구현
delete API 구현

solmoonkang committed:
fix: userInfo API update

solmoonkang committed:
fix: updatePassword API update

solmoonkang committed:
fix: delete API update

hongdosan commented on src/main/java/kr/co/itresumeregistersite/controller/UsersController.java:
```diff
@@ -15,31 +15,32 @@ public class UsersController {
     @GetMapping
-    public UsersInfoDto userInfo(Long userId) throws Exception {
-        return usersService.userInfo(userId);
+    public UsersInfoDto userInfo(String name) {
```
해당 회원 이름으로 조회를 하고 있는 것으로 보이는데, 같은 이름인 회원이 있으면 어떻게 하죠?

hongdosan replied:
그리고 @RequestParam이 없네요?

hongdosan commented on src/main/java/kr/co/itresumeregistersite/controller/UsersController.java:
```diff
@@ -15,31 +15,32 @@ public class UsersController {
     // 회원정보 수정
     @PutMapping
-    public void updateUser(UsersUpdateDto usersUpdateDto) throws Exception {
+    public void updateUser(UsersUpdateDto usersUpdateDto) {
```
@RequestBody는?

hongdosan commented on src/main/java/kr/co/itresumeregistersite/controller/UsersController.java:
```diff
@@ -15,31 +15,32 @@ public class UsersController {
+    // 회원 비밀번호 수정
     @PutMapping("/password")
-    public void updatePassword(UsersPasswordDto usersPasswordDto) throws Exception {
+    public void updatePassword(UsersPasswordDto usersPasswordDto) {
```
22

hongdosan commented on src/main/java/kr/co/itresumeregistersite/repository/UsersRepository.java:
```diff
@@ -8,5 +8,7 @@
 public interface UsersRepository extends JpaRepository<Users, Long> {
+
     Optional<Users> findByIdentity(String identity);
+    Optional<Users> findByName(String name);
```
이건 List로 받아와야 하는거 아닌지? 같은 이름인 회원이 있을 수도 있을텐데

hongdosan commented on src/main/java/kr/co/itresumeregistersite/service/UsersService.java:
```diff
@@ -32,59 +25,50 @@ public void signUp(SignUpDto signUpDto) {
-                .orElseThrow(() -> new UsersException(UsersExceptionType.NOT_FOUND_USERS));
+    @Transactional(readOnly = true)
+    public UsersInfoDto userInfo(String name) {
+        Users users = usersRepository.findByName(name)
```
22

hongdosan commented on src/main/java/kr/co/itresumeregistersite/service/UsersService.java:
```diff
@@ -32,59 +25,50 @@ public void signUp(SignUpDto signUpDto) {
-    public void updateUser(UsersUpdateDto usersUpdateDto) throws Exception {
+    public void updateUser(UsersUpdateDto usersUpdateDto) {
         Users users = usersRepository.findByIdentity(usersUpdateDto.getIdentity())
                 .orElseThrow(() -> new UsersException(UsersExceptionType.NOT_FOUND_USERS));
```
예외 클래스 혹은 타입 이름이 더욱 명시적이였으면 좋을 것 같네요.

solmoonkang replied:
UsersException과 UsersExceptionType 말하는 건가용?

hongdosan replied:
네~

HyuckJuneHong commented:
나쁘지 않네요

solmoonkang committed:
userInfo API findByName delete-> findByIdentity update
UsersRepository findByName delete

solmoonkang committed:
UsersException 경로 fix

solmoonkang committed:
build.gradle, application.properties - DB dependencies Add

solmoonkang committed:
findAllUserInfo API 구현
Controller userInfo @RequestParam Add
Controller updateUser, updatePassword @RequestBody Add

solmoonkang committed:
Controller userInfo @RequestParam fix

solmoonkang committed:
userInfo 회원 아이디 정보 조회로 fix

solmoonkang committed:
signUp @RequestBody Add

solmoonkang committed:
userInfo API fix

solmoonkang committed:
controller delete API @RequestBody Add

hongdosan commented on src/main/java/kr/co/itresumeregistersite/controller/UsersController.java:
```diff
@@ -1,45 +1,52 @@
     // 회원탈퇴
     @DeleteMapping
-    public void delete(DeleteDto deleteDto) throws Exception {
+    public void delete(DeleteDto deleteDto) {
```
앤 또 @RequestBody 어디갔노

hongdosan commented on src/main/java/kr/co/itresumeregistersite/service/UsersService.java:
```diff
@@ -32,66 +27,70 @@ public void signUp(SignUpDto signUpDto) {
+    // 전체 회원정보 조회
+    @Transactional(readOnly = true)
+    public List<Users> findAllUserInfo() {
+        return usersRepository.findAll();
```
?? 개인정보까지 다 보여주게? 비밀번호까지?

hongdosan commented on src/main/resources/application.properties:
```diff
@@ -1 +1,7 @@
-
+spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver
```
properties보다 YAML이 더 좋아요.

HyuckJuneHong commented:
전체적으로 좋아졌네요. 하지만 예외 처리 뿐만아니라 성공적으로 됐다면 잘 됐다는 알림도 잇으면 좋을 것 같아요. 
https://github.com/HyuckJuneHong/popool-server/blob/main/p-core/src/main/java/kr/co/popoolserver/error/model/ResponseFormat.java 참고

HyuckJuneHong commented:
테스트는 해보았나요?

solmoonkang committed:
build.gradle validation dependencies Add

solmoonkang committed:
findAllUserInfo API 생성 및 구현

solmoonkang committed:
fix : Exception class명 수정

solmoonkang committed:
fix : application.properties를 application.yml change

solmoonkang committed:
fix : findAllUserInfo API 수정

solmoonkang committed:
fix : findAllUserInfo API 수정

solmoonkang committed:
fix : findAllUserInfo API 수정

solmoonkang committed:
fix : UsersRepository findAll method delete

solmoonkang commented:
이번 피알 받고 테스트랑 DB연동까지 해보겠습니당

hongdosan commented on src/main/java/kr/co/itresumeregistersite/service/UsersService.java:
```diff
@@ -51,7 +51,8 @@ public UsersInfoDto userInfo(String identity) {
     public List<UsersInfoDto> findAllUserInfo() {
-        return usersRepository.findAll(Sort.by(Sort.Direction.ASC, "users_id")).stream()
+        return usersRepository.findAll(Sort.by(Sort.Direction.ASC, "users_id"))
+                .stream()
```
**이 로직이 어떤 것을 의미하는 걸까요?**

개인적으로 이전 리뷰를 이해못한 것 같아요.

HyuckJuneHong commented:
일단 어프로브 했으니, 머지하세요.
하지만, 전체 회원 정보 서비스는 더 이해를 하고 제대로 추후에 작성해야할 것 같아요.

아니면 저렇게 로직을 한 이유가 있을까요?

solmoonkang commented:
전체 회원 정보 서비스에 대해서 찾아보다가 저렇게 작성했습니당
추후에 다시 작성하도록 하겠습니다
